base_layer/service_framework/src/context/mod.rs

```diff
@@ -50,16 +54,18 @@ macro_rules! acquire_lock {
 }
 
 /// Simple collection for named handles
-#[derive(Default)]
-pub struct ServiceHandles {
-    handles: Mutex<HashMap<TypeId, Box<dyn Any + Sync + Send>>>,
+#[derive(Clone)]
+pub struct FinalServiceHandles {
```

Comment:
nit: I don't think there is a `NonFinalServiceHandles`, so I think `ServiceHandles` is fine. 

Suggestion: This is a bit of an AbstractFactory even. You could even add a member per Trait it exposes. E.g. 
`ServiceHandleFactory::get_local_node_interface() ->  &LocalNodeInterface`

Reply:
Agree - this was a product of figuring out what needs to change and I meant to change the name back.

The service framework is not coupled to core. It should stay that way. I think if we wanted to add a tari_core level abstraction that belongs in another PR.